src/CILogViewer.php

```diff
@@ -78,39 +63,30 @@ public function __construct() {
         $this->init();
     }
 
-
     /**
      * Bootstrap the library
      * sets the configuration variables
      * @throws \Exception
      */
     private function init() {
+        $viewerConfig = config('CILogViewer');
 
-        if(!function_exists("get_instance")) {
-            throw new \Exception("This library works in a Code Igniter Project/Environment");
+        if($viewerConfig) {
+            if(isset($viewerConfig->viewPath)) {
+                $this->viewPath = $viewerConfig->viewPath;
+            }
+            if(isset($viewerConfig->logFilePattern)) {
+                $this->logFilePattern = $viewerConfig->logFilePattern;
+            }
         }
-
-        //initiate Code Igniter Instance
-        $this->CI = &get_instance();
-
         //configure the log folder path and the file pattern for all the logs in the folder
-        $this->logFolderPath =  !is_null($this->CI->config->item(self::LOG_FOLDER_PATH_CONFIG_KEY)) ? rtrim($this->CI->config->item(self::LOG_FOLDER_PATH_CONFIG_KEY), "/") : rtrim(APPPATH, "/") . "/logs";
-        $this->logFilePattern = !is_null($this->CI->config->item(self::LOG_FILE_PATTERN_CONFIG_KEY)) ? $this->CI->config->item(self::LOG_FILE_PATTERN_CONFIG_KEY) : "log-*.php";
-
-        //concatenate to form Full Log Path
-        $this->fullLogFilePath = $this->logFolderPath . "/" . $this->logFilePattern;
-
-        //create the view file so that CI can find it
-        //use VIEWPATH constant so the CI can find views location and this constant will be defined in index.php file.
-        $this->LOG_VIEW_FILE_FOLDER = VIEWPATH . "cilogviewer";
-        $this->LOG_VIEW_FILE_PATH = rtrim($this->LOG_VIEW_FILE_FOLDER) . "/" . $this->LOG_VIEW_FILE_NAME;
-        if(!file_exists($this->LOG_VIEW_FILE_PATH)) {
-
-            if(!is_dir($this->LOG_VIEW_FILE_FOLDER))
-                mkdir($this->LOG_VIEW_FILE_FOLDER);
-
-            file_put_contents($this->LOG_VIEW_FILE_PATH, file_get_contents($this->LOG_VIEW_FILE_NAME, FILE_USE_INCLUDE_PATH));
+        $loggerConfig = config('Logger');
+        if(isset($loggerConfig->path)) {
```

Comment:
We should do this before line 74. This will ensure by default, we're using the default path. However, if the user provides an override via the config, we will honour it.